project-koku/koku
# COST-2356 - catch invalid payload and log better error message

Fixes [COST-2356](https://issues.redhat.com/browse/COST-2356)

Changes proposed in this PR:
* Do not run report summarization for an invalid payload because it will fail anyway.
* Log an error message that contains a portion of the `data_collection_message` if it exists
* Log the full cr_status as info when the payload is invalid

Testing Instructions:
1. Spin up in ephemeral
2. port forward to the sources service:
```
oc port-forward service/sources-api-svc 3000:8000
```
3. create source
```
curl --location --request POST 'http://localhost:3000/api/sources/v3.1/bulk_create' \
--header 'x-rh-identity: eyJpZGVudGl0eSI6IHsiYWNjb3VudF9udW1iZXIiOiAiMTAwMDEiLCAidHlwZSI6ICJVc2VyIiwgInVzZXIiOiB7InVzZXJuYW1lIjogInVzZXJfZGV2IiwgImVtYWlsIjogInVzZXJfZGV2QGZvby5jb20iLCAiaXNfb3JnX2FkbWluIjogZmFsc2V9fSwgImVudGl0bGVtZW50cyI6IHsiY29zdF9tYW5hZ2VtZW50IjogeyJpc19lbnRpdGxlZCI6IHRydWV9LCAic291cmNlcyI6IHsiaXNfZW50aXRsZWQiOiB0cnVlfX19' \
--header 'x-rh-sources-psk: thisMustBeEphemeralOrMinikube' \
--header 'Content-Type: application/json' \
--data-raw '{
    "sources": [{
        "name":"username - ocp",
        "source_ref": "582dfcce-898c-43de-83d0-7890d46caef6",
        "source_type_name": "openshift"
    }],
    "endpoints":[],
    "authentications":[],
    "applications":[{
        "application_type_id":"2",
        "source_name": "username - ocp"
    }]
}'
```
4. port forward ingress:
```
oc port-forward service/ingress-service 8000:8000
```
5. Send the attached payload to ingress:
```
curl -vvvv -F "file=@test-payload.gz;type=application/vnd.redhat.hccm.tar+tgz" -H "x-rh-identity: eyJpZGVudGl0eSI6IHsiYWNjb3VudF9udW1iZXIiOiAiMTAwMDEiLCAidHlwZSI6ICJVc2VyIiwgImludGVybmFsIjogeyJvcmdfaWQiOiAiMTAwMDEifSwgInVzZXIiOiB7InVzZXJuYW1lIjogInVzZXJfZGV2IiwgImVtYWlsIjogInVzZXJfZGV2QGZvby5jb20iLCAiaXNfb3JnX2FkbWluIjogdHJ1ZX19LCAiZW50aXRsZW1lbnRzIjogeyJjb3N0X21hbmFnZW1lbnQiOiB7ImlzX2VudGl0bGVkIjogdHJ1ZX0sICJzb3VyY2VzIjogeyJpc19lbnRpdGxlZCI6IHRydWV9fX0=" http://localhost:8000/api/ingress/v1/upload
```
6. Examine the listener logs. See the following:
```
[2022-02-23 23:22:19,313] INFO None {'message': 'Processing: /var/tmp/masu/insights_local/582dfcce-898c-43de-83d0-7890d46caef6/20220201-20220301/582dfcce-898c-43de-83d0-7890d46caef6_openshift_usage_report.0.csv complete.', 'tracing_id': '582dfcce-898c-43de-83d0-7890d46caef6'}
[1;31m[2022-02-23 23:22:19,316] ERROR None {'message': 'data collection error [operator]: error: query: sum(kube_pod_container_resource_limits{***} * on(pod, namespace) group_left kube_pod_status_phase{***}) by (pod, namespace, node): error querying prometheus: execution: found duplicate series for the match group {***} on the right hand-side of the operation: [{***}, {***}];many-to-many matching not allowed: matching labels must be unique on one side', 'tracing_id': '582dfcce-898c-43de-83d0-7890d46caef6', 'account': 'acct10001', 'provider_uuid': UUID('e370bb24-0d7f-406f-afcd-b21f14d494fa')}[0m
[2022-02-23 23:22:19,316] INFO None {'message': 'CR Status for invalid manifest: {"clusterID": "582dfcce-898c-43de-83d0-7890d46caef6", "clusterVersion": "stable-4.8", "api_url": "https://cloud.redhat.com", "authentication": {"type": "token", "credentials_found": true}, "packaging": {"last_successful_packaging_time": "2022-02-23T12:10:10Z", "max_reports_to_store": 30, "max_size_MB": 100, "number_reports_stored": 0}, "upload": {"ingress_path": "/api/ingress/v1/upload", "upload": true, "upload_wait": 8, "upload_cycle": 360, "last_upload_status": "202 Accepted", "last_payload_name": "20220223T121008-cost-mgmt.tar.gz", "last_payload_manifest_id": "582dfcce-898c-43de-83d0-7890d46caef6", "last_payload_request_id": "4728e62aef474cc7be80bac3a353f82e", "last_payload_files": ["582dfcce-898c-43de-83d0-7890d46caef6_openshift_usage_report.0.csv"], "last_successful_upload_time": "2022-02-23T12:10:34Z", "validate_cert": true}, "operator_commit": "77ec351f8d332796dc522e5623f1200c2fab4042", "prometheus": {"prometheus_config...
[2022-02-23 23:22:19,316] INFO None {'message': 'Sending Ingress Service confirmation for....
```

[test-payload.gz](https://github.com/project-koku/koku/files/8128480/test-payload.gz)

---
Additional Context:
